### PR TITLE
Java 11 support

### LINKED ISSUE #1004: Upgrade to java 9
[ ] tools.jar removed
   https://docs.oracle.com/javase/9/migrate/toc.htm#JSMIG-GUID-A868D0B9-026F-4D46-B979-901834343F9E****

### LINKED ISSUE #1366: Support for java 11
It's not clear if things work on Java 11 (as a deployment runtime)

### PR DESCRIPTION
**Pull request by mprins**
This brings both compile time and runtime Java 11 support.

- [x] expand AppVeyor and Travis-CI build matrix with Java 11
- [x]  set up testing on the Jenkins server with Java 11
- [x] upgrade GeoTools to version 21.0
- [x] upgrade b3p-commons-csw  https://github.com/B3Partners/b3p-commons-csw/pull/18
- [x] exclude Java 6 `tools.jar` system scoped transitive dependency from solr-core
- [x] fix compiler error due to usage of package `sun.security.rsa ` in `nl.b3p.viewer.admin.stripes.CycloramaConfigurationActionBean` (package sun.security.rsa is declared in module java.base, which does not export it to the unnamed module) #1374
- [x] fix compiler error due to usage of package of `com.sun.imageio.plugins.png` in  `nl.b3p.viewer.image.ImageTool` (package com.sun.imageio.plugins.png is declared in module java.desktop, which does not export it)
- [x] create an archive with libs that should be added when running on java 11 because these are no longer available in the runtime. eg.  `javax.xml.bind:jaxb-api`, `org.glassfish.jaxb:jaxb-runtime`, `javax.activation:javax.activation-api`,... see [jdk11+ profile](https://github.com/flamingo-geocms/flamingo/blob/7517f70acf1093a96d02221cda063538b81511b8/pom.xml#L903-L930) in the pom files
- [x] test deployment on Java 11 with recent tomcat
- [x] double check for conflicting/duplicate libraries in webapp/WEB-INF/lib and provided java 11 libs
- [x] Java EE packaging changes #1424

**note** in 33882dfcbdcbe3964d5d1bdabdcf4e4feb5237a5 I've added a patched version of the `gt-wms` module to work around https://osgeo-org.atlassian.net/browse/GEOT-6260 see also the comments in the pom file; see #1372

**note** after installing java 11 on Ubuntu you can easily switch in your console using
```bash
sudo update-alternatives --config java
export JAVA_HOME=/usr/lib/jvm/java-11-oracle
```

close #1366
close #1004

### CODE REVIEW

**Comment by mprins** [comment]
There is a problem compiling on Java 11, the solr libs drag in hadoop-annotations which hava system-scope dependency on tools.jar which is no longer available in java 11.
```
[INFO] +- org.apache.solr:solr-core:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-codecs:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-analyzers-kuromoji:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-analyzers-phonetic:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-highlighter:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-memory:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-misc:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-join:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-queryparser:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-spatial:jar:4.6.0:compile
[INFO] |  |  \- com.spatial4j:spatial4j:jar:0.3:compile
[INFO] |  +- org.apache.lucene:lucene-suggest:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-grouping:jar:4.6.0:compile
[INFO] |  +- org.apache.lucene:lucene-queries:jar:4.6.0:compile
[INFO] |  +- com.carrotsearch:hppc:jar:0.5.2:compile
[INFO] |  +- com.googlecode.concurrentlinkedhashmap:concurrentlinkedhashmap-lru:jar:1.2:compile
[INFO] |  +- commons-cli:commons-cli:jar:1.2:compile
[INFO] |  +- joda-time:joda-time:jar:2.2:compile
[INFO] |  +- org.apache.hadoop:hadoop-annotations:jar:2.2.0:compile
[INFO] |  |  \- jdk.tools:jdk.tools:jar:1.6:system
```

~~not sure what happens if we~~ exclude the system scoped tools .jar yield no problems

File: viewer/src/main/java/nl/b3p/viewer/image/ImageTool.java [hunk 1]
```diff
@@ -124,21 +123,9 @@ public static BufferedImage readImage(HttpMethod method, String mime) throws Exc
...
-                    for (int x = 0; x < i.getWidth(); x++) {
-                        for (int y = 0; y < i.getHeight(); y++) {
-                            int rgb = i.getRGB(x, y);
-                            rgb = (rgb & 0xFFFFFF) == alphaPix ? alphaPix : rgb;
-                            tmp.setRGB(x, y, rgb);
-                        }
-                    }
-                    i = tmp;
-                }
+            if (!i.getColorModel().hasAlpha() && mimeType.toLowerCase().contains("png")) {
```
Comment (mprins) [inline viewer/src/main/java/nl/b3p/viewer/image/ImageTool.java:126]: :+1:

**Comment by mprins** [comment]
@mtoonen I think this is good to go

**Comment by mtoonen** [comment]
Installed java11, will try it out

**Review by mtoonen — APPROVED** [review]
Seems to work

**Pull request merged**